src/builtins/commandline.cpp

```diff
@@ -167,6 +168,7 @@ maybe_t<int> builtin_commandline(parser_t &parser, io_streams_t &streams, const
                                                   {L"line", no_argument, nullptr, 'L'},
                                                   {L"search-mode", no_argument, nullptr, 'S'},
                                                   {L"paging-mode", no_argument, nullptr, 'P'},
+                                                  {L"paging-full-mode", no_argument, nullptr, 'F'},
```

Comment:
I considered giving the existing `-P/--paging-mode` an optional argument, like `--paging-mode=full`, but that would work slightly worse with interactive completions, so yours seems fine.

Maybe just remove the short option, since it's not so obvious?